src/include/platform/OTARequestorDriver.h

```diff
@@ -93,6 +99,25 @@ class OTARequestorDriver
 
     /// Called when the current software update has been cancelled by the local application
     virtual void UpdateCancelled() = 0;
+
+    /// Inform the driver that the device commissioning has completed
+    virtual void OTACommissioningCallback() = 0;
+
+    using ProviderLocationType = app::Clusters::OtaSoftwareUpdateRequestor::Structs::ProviderLocation::Type;
+    virtual void
+    /// Driver portion of the logic for processing the AnnounceOTAProviders command
+    ProcessAnnounceOTAProviders(const ProviderLocationType & providerLocation,
+                                app::Clusters::OtaSoftwareUpdateRequestor::OTAAnnouncementReason announcementReason) = 0;
+
+    /// Direct the driver to trigger the QueryImage command. The driver may choose to execute some internal
+    /// logic and will then call an OTARequestor API to actually send the command. The purpose of this
+    /// function is to allow implementation-specific logic (such as possibly cancelling an ongoing update)
+    /// to be executed before triggering the image update process
+    virtual void SendQueryImage() = 0;
+
+    // Returns the next available Provider location
```

Comment:
Can you expand this comment to describe what "next available" means here? Is it walking the DefaultOTAProviders list?

Reply:
Done